### PR TITLE
Various fixed for DPCU 1.5 specs + cheap DVI/DIV

### PR DESCRIPTION
**Pull request by ghost**
Various little changes with a test case each time.

### CODE REVIEW

File: lib/Target/DCPU16/DCPU16InstrInfo.td [hunk 1]
```diff
@@ -384,10 +384,12 @@ let Defs = [REX] in {
   defm SRL16   : BASIC_RR_NON_COM<0x0, "SHR", srl>,  BASIC_NORMAL<0x0, "SHR", srl>;
   defm SRA16   : BASIC_RR_NON_COM<0x0, "ASR", sra>,  BASIC_NORMAL<0x0, "ASR", sra>;
   defm SHL16   : BASIC_RR_NON_COM<0x0, "SHL", shl>,  BASIC_NORMAL<0x0, "SHL", shl>;
+  defm UREM16  : BASIC_RR_NON_COM<0x0, "MOD", urem>, BASIC_NORMAL<0x0, "MOD", urem>;
```
Comment (Blei) [inline lib/Target/DCPU16/DCPU16InstrInfo.td:386]: Now we have MOD twice. Maybe remove the MOD16 instruction?

**Comment by Blei** [comment]
I was _just_ about to submit a nearly identical pull request ;) Great that you found the switch to turn of the x/2 optimization! LGTM.

File: lib/Target/DCPU16/DCPU16InstrInfo.td (hunk 1 above)
Comment (ghost) [inline lib/Target/DCPU16/DCPU16InstrInfo.td:386]: Upps. Let me fix that.

**Comment by ghost** [comment]
We should probably use the assign function in the issues to assign task to us. That way we won't waste precious time ;)

**Pull request merged**